src/vs/workbench/services/untitled/common/untitledTextEditorModel.ts

```diff
@@ -365,7 +365,8 @@ export class UntitledTextEditorModel extends BaseTextEditorModel implements IUnt
 		}
 
 		if (modelFirstWordsCandidate !== this.cachedModelFirstLineWords) {
-			this.cachedModelFirstLineWords = modelFirstWordsCandidate;
+			// Normalize multiple whitespaces to a single whitespace for the untitled editor label
+			this.cachedModelFirstLineWords = modelFirstWordsCandidate ? modelFirstWordsCandidate.replace(/\s+/g, ' ') : '';
```

Comment:
I would put this after the `.trim()` call here:

https://github.com/microsoft/vscode/blob/50d9f05334aee7bbc416f675cceccd9a174de8c7/src/vs/workbench/services/untitled/common/untitledTextEditorModel.ts#L362-L362

Because then we do a good check `modelFirstWordsCandidate !== this.cachedModelFirstLineWords`

Thanks!

Reply:
Update: remove extra spaces after trim first line text.